src/components/flat-table/flat-table-body-draggable/flat-table-body-draggable.component.js

```diff
@@ -0,0 +1,107 @@
+import React, { useEffect, useState, useRef } from "react";
+import { useDrop, DndProvider } from "react-dnd";
+import Backend from "react-dnd-html5-backend";
+import PropTypes from "prop-types";
+import StyledIcon from "../../icon/icon.style";
+
+import FlatTableBody from "../flat-table-body/flat-table-body.component";
+import FlatTableCell from "../flat-table-cell/flat-table-cell.component";
+
+const DropTarget = ({ children, getOrder, ...rest }) => {
+  const [, drop] = useDrop({
+    accept: "flatTableRow",
+    drop() {
+      getOrder();
+    },
+  });
+
+  return (
+    <FlatTableBody ref={drop} {...rest}>
+      {children}
+    </FlatTableBody>
+  );
+};
+
+const FlatTableBodyDraggable = ({ children, getOrder }) => {
+  const [draggableItems, setDraggableItems] = useState(
+    React.Children.toArray(children)
+  );
+  const isFirstRender = useRef(true);
+  useEffect(() => {
+    if (!isFirstRender.current) {
+      setDraggableItems(React.Children.toArray(children));
+    } else {
+      isFirstRender.current = false;
+    }
+  }, [children]);
+
+  const findItem = (id) => {
+    const draggableItem = draggableItems.filter((item) => {
```

Comment:
Not blocking the PR but you could just inline this function and remove the `return` etc

Reply:
It's needed in the flat-table-row component and is used by the drag handler to give scope of an row's id within the table